src/components/Personalization/createOnDecisionHandler.js

```diff
@@ -10,6 +10,22 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
+import { MESSAGE_IN_APP } from "../../constants/schema.js";
+
+const createShouldSuppressDisplay = () => {
+  let count = 0;
+  return (proposition) => {
+    const { items = [] } = proposition;
+
+    if (!items.some((item) => item.schema === MESSAGE_IN_APP)) {
+      return false;
+    }
+    count += 1;
+
+    return count > 1;
```

Comment:
nitpick: a comment here on why we only suppress display after there has been at least 2 `items` with no in-app messages would be extremely helpful.